src/reader/mod.rs

```diff
@@ -459,73 +460,47 @@ pub struct Reader<R> {
 }
 
 /// Builder methods
-impl<R> Reader<R> {
+impl<R: Read> Reader<Utf8BytesReader<R>> {
     /// Creates a `Reader` that reads from a given reader.
     pub fn from_reader(reader: R) -> Self {
         Self {
-            reader,
+            reader: Utf8BytesReader::new(reader),
             parser: Parser::default(),
         }
     }
+}
 
-    configure_methods!();
+/// Builder methods
+impl<'a> Reader<&'a [u8]> {
+    /// Creates an XML reader from a string slice.
+    pub fn from_str(s: &'a str) -> Self {
+        // Rust strings are guaranteed to be UTF-8, so lock the encoding
+        #[cfg(feature = "encoding")]
+        {
+            let mut parser = Parser::default();
+            parser.encoding = EncodingRef::Explicit(UTF_8);
+            Self {
+                reader: s.as_bytes(),
+                parser: parser,
+            }
+        }
+
+        #[cfg(not(feature = "encoding"))]
+        {
+            Self {
+                reader: s.as_bytes(),
+                parser: Parser::default(),
+            }
+        }
+    }
 }
 
-/// Getters
+/// Public implementation-independent functionality
 impl<R> Reader<R> {
-    /// Consumes `Reader` returning the underlying reader
-    ///
-    /// Can be used to compute line and column of a parsing error position
```

Comment:
I'm thinking we really need a better solution than this.  Either by tracking newlines during parsing or by automatically implementing something like this for `&str` and perhaps `File`.

Reply:
I'm not sure about the overhead of the former approach, might be too expensive.